### PR TITLE
add rafu.cancelFrame

### LINKED ISSUE #1: method to cancel frame / throttle
@iamdustan suggested an system with an identifier, working something like

```js
rafu.cancel(frameOrThrottleId) {
  if (useRaf) {
    cancelAnimationFrame(frameOrThrottleId);
    return;
  }
  clearTimeout(frameOrThrottleId);
}
```

For reference, lodash 5's debounce/throttle will be based on rAF, when available: https://github.com/lodash/lodash/pull/3567/files

### PR DESCRIPTION
**Pull request by forresto**
I think this fixes #1, tests included...

### CODE REVIEW

**Comment by forresto** [comment]
@iamdustan @danro want to CR this public PR?

File: rafu.js [hunk 1]
```diff
@@ -8,38 +8,66 @@
...
     wait = true;
     context = this;
-    frame(function() {
+    rafOrTimeoutId = frame(function() {
       wait = false;
       func.apply(context, args);
     });
   };
+
+  throttled.cancelFrame = function() {
```
Comment (iamdustan) [inline rafu.js]: nit: rename this to just `cancel`. Lodash seems to provide both `cancel` and `flush` methods on the returned function.

https://lodash.com/docs/4.17.11#debounce

Screen Shot 2019-06-03 at 4 35 07 PM

File: tests.js [hunk 2]
```diff
@@ -38,6 +42,32 @@ describe('rafu', function() {
...
       });
     });
+
+    it('should return a numeric id > 0', function() {
+      var id = rafu.frame(function() {});
+      expect(id).to.be.greaterThan(0);
+    });
+  });
+
+  describe('cancelFrame', function() {
```
Comment (iamdustan) [inline tests.js]: nice tests! :clap:

File: rafu.js (hunk 1 above)
Comment (forresto) [reply rafu.js]: I mirrored `rafu.cancelFrame`... should they both be `cancel`?

Comment (forresto) [reply rafu.js]: Nevermind, `throttledFunc.cancel()` would make sense.

Comment (iamdustan) [reply rafu.js]: I’d personally just call them both `cancel`.

Comment (forresto) [reply rafu.js]: Changed name and added test that it's there: https://github.com/danro/rafu/pull/2/commits/254e3814bcc9ecd2310c4c380afd617023233cb9

Comment (forresto) [reply rafu.js]: Meh... `cancel(thing)` seems a little generic. 🤷‍♀

Comment (danro) [reply rafu.js]: I like `throttled.cancel()` as well. Generic isn't bad in this case, we simply want the throttled function to cancel the execution (without caring too much about the implementation details).

Comment (danro) [reply rafu.js]: `rafu.cancel` also seems more in-line with the simplicity of the module. If we supported other things besides frames then maybe the distinction would matter.

**Comment by forresto** [comment]
> some older browsers do not support cancelAnimationFrame even though they support the request function.

@danro Ah! TIL/WTF. How is this? https://github.com/danro/rafu/pull/2/commits/5ea40d97a62f9da9e681b07a4c68500cefc458e0

**Pull request merged**